pkg/urbit/vere/io/ames.c

```diff
@@ -324,144 +325,143 @@ _ames_ship_of_chubs(c3_d sip_d[2], c3_y len_y, c3_y* buf_y)
   memcpy(buf_y, sip_y, c3_min(16, len_y));
 }
 
-/* _ames_sift_prelude(): parse prelude, returning length
+/* _ames_sift_head(): parse packet header.
 */
-static c3_y
-_ames_sift_prelude(u3_head* hed_u,
-                   u3_prel* pre_u,
-                   c3_w     len_w,
-                   c3_y*    pre_y)
+static void
+_ames_sift_head(u3_head* hed_u, c3_y buf_y[4])
 {
-  c3_y rog_y, sen_y, rec_y, len_y;
+  c3_w hed_w = _ames_sift_word(buf_y);
 
-  rog_y = ( c3y == hed_u->rel_o )? 6 : 0;
-  sen_y = 2 << hed_u->sac_y;
-  rec_y = 2 << hed_u->rac_y;
-  len_y = rog_y + sen_y + rec_y + 1;
+  //  first three bits are reserved
+  //
+  hed_u->req_o = (hed_w >>  2) & 0x1;
```

Comment:
Off-by-one? (I realize it was already here.)

```suggestion
  //  first two bits are reserved
  //
  hed_u->req_o = (hed_w >>  2) & 0x1;
```